test/libethereum/blockchain.cpp

```diff
@@ -224,14 +284,15 @@ void doBlockchainTests(json_spirit::mValue& _v, bool _fillin)
 				stateOptionsMap expectStateMap;
 				State stateExpect(OverlayDB(), BaseState::Empty, biGenesisBlock.coinbaseAddress);
 				importer.importState(o["expect"].get_obj(), stateExpect, expectStateMap);
-				ImportTest::checkExpectedState(stateExpect, state, expectStateMap, Options::get().checkState ? WhenError::Throw : WhenError::DontThrow);
+				ImportTest::checkExpectedState(stateExpect, trueState, expectStateMap, Options::get().checkState ? WhenError::Throw : WhenError::DontThrow);
 				o.erase(o.find("expect"));
 			}
 
 			o["blocks"] = blArray;
-			o["postState"] = fillJsonWithState(state);
+			o["postState"] = fillJsonWithState(trueState);
+			o["lastblockhash"] = toString(trueBc.info().hash());
```

Comment:
in _fillin == false part, we should check that our last block has this hash (as we expect the other clients to do).